crates/astria-sequencer/src/app.rs

```diff
@@ -163,6 +174,7 @@ impl App {
         // clear the cache of transaction execution results
         self.execution_result.clear();
         self.processed_txs = 0;
+        self.proposal_executed = false;
```

Comment:
put this in `commit` instead - this function is only called by `prepare_proposal`/`process_proposal`, so if the node was a validator for the previous round but not the current round, then `proposal_executed` will be stuck being set to true

Reply:
nvm, you did put this in commit. this is fine then!